M3 design Completed

Aaron-RN opened the pull request:
Edited CSS classes to fit the specifications of Microverse's guidelines

Aaron-RN committed:
Update manifest.json

Aaron-RN committed:
Update README.md

Aaron-RN committed:
added Big.js and prop-types dependencies

Aaron-RN committed:
linter error fixes

Aaron-RN committed:
Added Multiple Components

Added Multiple Components
- Display
- Button Panel
- Button

Display shows  the result of the calculator's calculator

Button Panel renders and contains all buttons

Added CSS to shape the calculator

Aaron-RN committed:
Edited CSS

Added Screenshot

Aaron-RN committed:
Update App.css

Added max and min width to calculator class for responsive sizing

Aaron-RN committed:
Delete Screenshot.png

Aaron-RN committed:
Create screenshot.png

MauricioRobayo commented on src/components/button.js:
```diff
@@ -0,0 +1,30 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+function Button(props) {
+  const { onClick, name } = props;
```
Please make sure it accepts `color` and `wide` as an incoming props ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorAndWideProps.png)).

Optional: You can destructure the incoming props on the argument, and since it is a stateless component, you could use arrow functions:

```js
const Button = ({ onClick, name }) => (
  <button
    className={`btn${name === '0' ? ' btn-zero' : ''}`}
    type="button"
    onClick={onClick}
  >
    {name}
  </button>
);
```
But you'll have to change this to take `color` and `wide` props.

Aaron-RN replied:
Ahhhhhh cool! Noted.

MauricioRobayo commented on src/components/button.js:
```diff
@@ -0,0 +1,30 @@
+}
+
+// Specifies the default values for props:
+Button.defaultProps = {
+  name: null,
+  onClick: null,
+};
```
Default `color` for button is orange   ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorProps.png)).

`wide` is boolean, true means that button take 50% of space (used for `0` button)   ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/WideProps.png)).

Aaron-RN replied:
Hah, I read that but I thought, but only 5 out of the 10 + buttons are orange... So why isn't grey the default.

MauricioRobayo reviewed (changes requested):
What are you doing here? Get out!

![Peek 2020-04-08 18-33](https://user-images.githubusercontent.com/2121481/78843138-94325200-79c7-11ea-8666-8fc9ca88de83.gif)

This is pure awesomeness. I was tired and about to call it a day and picked one last review, and this came up... you made my day!

But... I cannot approve of it 😞. Please double-check the following requirement for the `Button` component:

* [x] Button component accepts `color` and `wide` as an incoming props   ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorAndWideProps.png))
* [x] default `color` for button is orange   ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/colorProps.png))
* [x] `wide` is boolean, true means that button take 50% of space (used for `0` button)   ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone3/WideProps.png))

Since I can not approve of it but I think this is so great, here you have a gif, you deserve it 😃:

![this-is-awesome](https://user-images.githubusercontent.com/2121481/78844176-a06bde80-79ca-11ea-9a94-84efa4ee17bf.gif)

Aaron-RN committed:
Button Component and Button PanelComponent Updated

- Add - Button component now accepts two more props from Button Panel Component.
- Props = Color and Wide. Default color is now orange.

- Update - CSS for buttons changed

Aaron-RN committed:
Update display.js

Converted Display function to ES6 function and imbedded value prop in argument using object destructuring

Aaron-RN committed:
Linter fix

shloch reviewed (approved):
## hello @Aaron-RN

We're good to go.....great job

![accomplished](https://user-images.githubusercontent.com/40712459/78934094-43435c00-7aa2-11ea-8d2c-1ceab85dbc33.jpg)

### on to the next milestone...we gooo !
